pkg/fanal/applier/docker.go

```diff
@@ -232,6 +234,13 @@ func ApplyLayers(layers []ftypes.BlobInfo) ftypes.ArtifactDetail {
 		}
 	}
 
+	// De-duplicate same debian packages from different dirs
+	// cf. https://github.com/aquasecurity/trivy/issues/8297
+	packages := lo.UniqBy(mergedLayer.Packages, func(pkg ftypes.Package) string {
+		return cmp.Or(pkg.ID, fmt.Sprintf("%s@%s", pkg.Name, utils.FormatVersion(pkg)))
+	})
+	mergedLayer.Packages = lo.Ternary(len(packages) > 0, packages, nil)
```

Comment:
Since this conversion is needed in many places, we may want to define a utility function somewhere, like:

```go
func NilIfEmpty[T any](s []T) []T {
    if len(s) == 0 {
        return nil
    }
    return s
}
```

Reply:
hm.. we already have this function 😄 :
https://github.com/aquasecurity/trivy/blob/60491f8a7e4a706c8a78586c010932ff3756c691/pkg/x/slices/slices.go#L3-L8

Used this function in https://github.com/aquasecurity/trivy/pull/8298/commits/4f939b793a879fc398683bfb3503142b2479191b

Reply:
Oh, we already have it 😄